src/main/java/com/aws/greengrass/telemetry/SystemMetricsEmitter.java

```diff
@@ -40,6 +54,58 @@ public void emitMetrics() {
         }
     }
 
+#if ANDROID
+    private long getUsedMemory() {
+        long usedMemory = -1;
+        try {
+            MemoryInfo mi = new MemoryInfo();
+            Class activityThreadClass = Class.forName("com.aws.greengrass.nucleus.androidservice.MainActivity");
+            Field contextField = activityThreadClass.getDeclaredField("context");
+            Context ctx = (Context) contextField.get(null);
+            ActivityManager activityManager = (ActivityManager) ctx.getSystemService(Context.ACTIVITY_SERVICE);
+            activityManager.getMemoryInfo(mi);
+            usedMemory = mi.totalMem - mi.availMem;
+        } catch (Exception ex) {
+            logger.atInfo().log(ex);
+        }
+        return usedMemory;
+    }
+
+    private long getOpenFileDescriptorsCount() {
+        /* We are starting from -1. Even if command will be executed correctly, the first line of
+           responce is header and should not be counted. */
+        long openFDCount = -1;
+        try {
+            /* lsof returns list of open file descriptors for current process */
+            java.lang.Process proc = Runtime.getRuntime().exec(
+                    "/system/bin/lsof -p ".concat(String.valueOf(android.os.Process.myPid())));
+            BufferedReader stdInput = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+            while (stdInput.readLine() != null) {
+                openFDCount += 1;
+            }
+        } catch (Exception ex) {
+            logger.atInfo().log(ex);
+        }
+        return openFDCount;
+    }
+
+    private double getCpuLoad() {
+        double cpuLoad = -1.;
+        try {
+            Class activityThreadClass = Class.forName("com.aws.greengrass.nucleus.androidservice.MainActivity");
+            Field contextField = activityThreadClass.getDeclaredField("context");
+            Context ctx = (Context) contextField.get(null);
+            HardwarePropertiesManager hardwarePropertiesManager = ctx.getSystemService(HardwarePropertiesManager.class);
+            /* FIXME: Rework resulting data when nucleus will be system service with required permissions */
+            CpuUsageInfo[] cpuUsages = hardwarePropertiesManager.getCpuUsages();
+        } catch (Exception ex) {
+            /* FIXME: Change atDebug with atInfo when nucleus will be system service */
+            logger.atDebug().log(ex);
+        }
+        return cpuLoad;
+    }
+#endif /* ANDROID */
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Syntax error on token &#34;Invalid Character&#34;, @ expected

This file contains syntax errors and can&#39;t be analyzed by all of CodeGuru Reviewer&#39;s detectors. Fix the syntax errors and run the repository analysis or submit the pull request again.